### PR TITLE
[hotfix] Block Spam Email Domains [OSF-6976]

### PR DESCRIPTION
**Pull request by caseyrollins**
#### Purpose
- Prevents users from creating an account with a spam email domain. Additionally, talks/posters cannot be submitted to conferences/meetings from a blacklisted email address, and users cannot add blacklisted email addresses as alternate emails.

#### Changes
- Adds an _extensive_ list of blacklisted domains to `defaults.py`. 

#### Side effects
- None expected.

#### Ticket
- [OSF-6976](https://openscience.atlassian.net/browse/OSF-6976)

### CODE REVIEW

File: framework/auth/core.py [hunk 1]
```diff
@@ -498,6 +498,9 @@ def create_unregistered(cls, fullname, email=None):
 
     @classmethod
     def create(cls, username, password, fullname):
+        if username.split('@')[1] in settings.BLACKLISTED_DOMAINS:
```
Comment (icereval) [inline framework/auth/core.py]: Any reason not to just implement the `validate_email` helper from `framework/auth/utils.py`?

File: website/settings/defaults.py [hunk 2]
```diff
@@ -563,3 +563,1173 @@ def parent_dir(path):
...
+    'zetmail.com',
+    'zippymail.info',
+    'zoaxe.com',
+    'zoemail.com',
+    'zoemail.net',
+    'zoemail.org',
+    'zomg.info',
+    'zxcv.com',
+    'zxcvbnm.com',
+    'zzz.com'
```
Comment (icereval) [inline website/settings/defaults.py]: ending `,`

File: framework/auth/core.py (hunk 1 above)
Comment (mfraezz) [inline framework/auth/core.py]: Seconded

**Comment by mfraezz** [comment]
Per request of @icereval, closing in favor of #6261 (contains small update). Consider this merged 🗽

**Pull request closed without merging**